src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -62,19 +62,20 @@ join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeS
     {ok, Membership, OutgoingMessages, State}.
 
 try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node) ->
+    Dist = 0,
```

Comment:
This is going to cause a problem: you're pulling Dist out of the record and then on L65 you're saying Dist = 0 -- if Dist is actually 0, then this will succeed, otherwise, this will crash if the values aren't the same.

Reply:
Ok, I've gotten rid of that, sorry. But I don't understand why it's saying 'Dist' in unbound because it's in the record and I put it in the function call each time I used it.

Reply:
That error is coming from L21.  

If you look at L21, you're trying to put Dist inside of a record.  However, you haven't defined Dist.  Dist is unbound -- it doesn't have a value yet.